### ISSUE TITLE
Couldn’t change tracks on empeg after starting car and using Siri (rare)

### ISSUE THREAD

**Issue by tfabris** [tfabris/BlueGigaEmpeg#65]
Had a rare situation where AVRCP was definitely working, audio worked, and track titles were updating, but thge empeg didn’t respond to the "next track" command or any other command for that matter. 

It occurred after I had made a text message using Siri Voice Recognition on my iPhone which was also paired to bluetooth at the time. It was at the start of a drive and I needed to send a text and complete it before I started moving. I pressed the steering wheel button for Siri and the car switched over to Siri, muting/pausing the empeg in the process. After completing the text and Siri disconnected, then the Empeg was no longer responding to any of the steering wheel or touch screen controls.

The interesting part was that the track titles on the car stereo screen were still working. I could press an empeg front-panel button and the track text would change on the car stereo touchscreen. So AVRCP was working, it was receiving data *from* the empeg successfully still. It was only commands which should have been sent *to* the empeg which were not working.

Sleep/wake empeg (reboot bluetooth module by bouncing the power by sleeping/waking the empeg) fixed the issue. So it wasn’t a physically bad serial connection such as loose wires or loose plug.

Suspicion: somehow the BGE missed the line of the empeg output that told it the player was runnIng. 

Possible fix: the detection of the player being awake and running needs to have more messages that it can use to perform that detection. Not just the prolux message.

**Comment by tfabris** [comment]
Looked at the existing code and it’s not due to the code mischaracterizing the player state.  The code already has a feature for this. 

My next guess is an avrcp mixup with registrations and message sequence ids. However I can’t currently imagine a scenario where this could occur. 

Attempted repro of issue with Siri engagement and it didn’t repro.

**Comment by tfabris** [comment]
Got a second repro, though the debug cable was not connected at the time. Conditions (not sure which conditions are the required conditions for the repro):

- car is cold.
- car in Work garage. 
- start car.
- Bluetooth pairs successfully with empeg and with phone. 
- empeg plays music gud 
- no toucha da empeg
- no toucha da next track buttonz 
- press hold Siri button on steering wheel and make a phone call. 
- empeg successfully pauses during phone call
- when done phone Call hangs up by itself. 
- after back at main screen try next track button. 
- no worky 
- press next on empeg front panel, worky, new track title worky
- steering wheel and touchscreen no worky
- switch to FM radio and back still no worky
- press stereo off button and empeg does not pause as it should do

**Comment by tfabris** [comment]
Sleep wake empeg (reboot Bluetooth) = worky

**Comment by tfabris** [comment]
Sleep wake empeg reboot Bluetooth then try Siri call = no repro, worky

**Comment by tfabris** [comment]
Power off car completely power on car without engine running

Same steps as before

No repro, worky

**Comment by tfabris** [comment]
I’ve tried several attempts to repro on the debug cable and can’t get a repro. 

I don’t think I’m Heisenberging it with the debug cable. Since the repro is rare and intermittent I think it’s just coincidence. I’ve tried many repros both with and without the debug cable and I’ve only had two repros so far, both of them off-cable.

**Comment by tfabris** [comment]
I had a third repro. It was unfortunately without the debug connection, and was just a short time after I had attempted a repro with the debug connection connected.

My recollection of this repro was that it did NOT require engaging Siri. It was just getting in the car turning it on, cold start. All other symptoms were the same.

**Comment by tfabris** [comment]
I got a fourth repro of this (also not on the debug cable) but something interesting this time:

- Issue occurred immediately after starting car. Engaging Siri was not required to cause the bug to reproduce.
- Track title changes definitely still worked correctly; I could change tracks on the empeg front panel and they changed on the car stereo screen. Yet still I could not change tracks or play/pause with the touchscreen or steering wheel. 
- But here's what's interesting. *After* I had confirmed the above behaviors and I had a repro, *THEN* I engaged Siri with the Siri button on my car steering wheel. And you know what happened? THE EMPEG PAUSED.

This tells me that the car stereo headunit has no problem sending commands to the BlueGigaEmpeg when it feels like it wants to do so. All the messaging and handshaking *WORKS* as designed. The only issue is that sometimes my car stereo boots up and it doesn't feel like sending commands to my empeg.

I think this is a Honda bug and there's nothing I can do about it. Closing bug until I think of some way that the above thing could possibly be my fault.
